roles/ckan/templates/kubernetes/ckan.yaml

```
@@ -86,6 +86,10 @@ spec:
               value: "/usr/lib/ckan/.minikubevenv"
             - name: XDG_CACHE_HOME
               value: "/usr/lib/ckan/.minikubevenv/cache"
+{% if ckan_googleanalytics_enable %}
+            - name: CKAN_GOOGLEANALYTICS_ENABLED
+              value: "1"
+{% endif %}
```

```suggestion
            - name: CKAN_GOOGLEANALYTICS_ENABLED
              value: "{{ ckan_googleanalytics_enable }}"
```

Maybe this? (I find it much easier to follow this file if it's inline where possible!

---

Also is nested inside `{% if fjelltopp_env_type == 'local' %}` whereas I assume we want it for all envs

---

With regard to the suggested change, it won't work because the [code](https://github.com/fjelltopp/who-afro-ckan/blob/267-cannot-not-enable-google-analytics/ckan-entrypoint.sh#L72-L77) that uses the environment variable merely checks for its existence, not its actual value.  
So the idea is to only create that environment variable if Google analytics is enabled.

With regard to the nesting, the bootstrapping seems to happen only in the init container. In fact this is something I wanted to touch base with you about because I couldn't understand why you were seeing this issue in production.
So yeah, we probably want this in all envs but I need to appreciate the subtleties  first before commit code for all envs.

---

The ckan-entrypoint you are changing in your other PR gets built into the docker image. The docker image is what is used for cloud deployments.

Also, why not change fjelltopp/who-afro-ckan#54 to look at the value rather than for existence. Generally, in my experience, just checking for existence is less robust than checking for an explicit value?

---

@ChasNelson1990 @cooper667 I have done as per your feedback.